Clapet/Views/Settings/GeneralSettings.swift

```diff
@@ -90,6 +100,14 @@ struct GeneralSettings: View {
         .padding(.horizontal, 20)
     }
     
+    func toggleDockIcon() {
+        if showDockIcon {
+            NSApp.setActivationPolicy(.regular)
+        } else {
+            NSApp.setActivationPolicy(.accessory)
+        }
+    }
+    
```

Comment:
I believe this should not be handled here 
- The activation policy is already regular when in the settings so this is redundant, and it instantly hides the settings when you untick the checkbox
- When you restart the application, the setting is not taken into account  and the dock icon is not visible

Instead, nothing should be done here and we should : 
- not call `AppDelegate.hideApplication()` on launch if `showDockIcon` is true (the code that handles hiding the application on launch is in the `checkForUpdate` function, I don't know why I did that, it's not very intuitive)
- do the thing in the other comment about SettingsView
- Call `AppDeletage` hide and show functions instead of NSApp for clarity